go-controller/pkg/node/node_ip_handler_linux.go

```diff
@@ -139,6 +139,22 @@ func (c *addressManager) ListAddresses() []net.IP {
 	return out
 }
 
+func (c *addressManager) ListNetworkAddresses() []*net.IPNet {
+	c.Lock()
+	defer c.Unlock()
+	addrs := sets.List(c.cidrs)
+	out := make([]*net.IPNet, 0, len(addrs))
+	for _, addr := range addrs {
+		_, cidr, err := net.ParseCIDR(addr)
+		if err != nil {
+			klog.Errorf("Failed to parse %s: %v", addr, err)
+			continue
+		}
+		out = append(out, cidr)
+	}
+	return out
```

Comment:
I understand that this is whow it's done at `ListAddresses` but this should be done when c.cidr is set and store at the addresssManager as `set.Set<*net.IPNet>` or `[]*net.IPNet`, it do not make sense to do this list to set + parse at every retrievel.

Also we should use here a read/write lock instead of a normal lock.

Just ranting, up to you for a commit refactoring this.

Reply:
aah I see what you mean. I admit it's not the most elegant solution, but I could only find uglier ways :| 
Since there are customers waiting for this fix, maybe we can postpone this to my followup PR (https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5093) ?

Reply:
I am ok with follow up, thanks!

Reply:
I'm unsure here, but I'll defer to a maintainer.

Reply:
I don't think we need a RWMutex here yet. However, I think maybe an easy optimization is that ListNetworkAddresses seems to be called everywhere ListAddresses is called. In ListAddresses we do the same parsing, why not just make ListAddresses return ([]net.IP, []*net.IPNet)? That will make this PR not incur any perf regression.